### PR TITLE
Bump JupyterBook to 1.0, unpin python and jupyterlab

### LINKED ISSUE #123: JupyterBook 1.0 upgrade
https://jupyterbook.org/en/stable/explain/migration.html

Also, would be interesting to try this extension, which would make notebooks in jupyterlab appear as they do on the rendered website with admonitions and all:
https://github.com/executablebooks/jupyterlab-myst

### PR DESCRIPTION
**Pull request by scottyhq**
Closes #123 

I think we should for the most part *unpin* dependencies in environment.yml, since they are explicitly pinned whenever we create a lockfile.

### CODE REVIEW

File: conda/environment.yml [hunk 1]
```diff
@@ -8,13 +8,13 @@ channels:
...
   - ipyleaflet
-  - jupyter-book=0.13
+  - jupyter-book
   - jupyter-resource-usage
   - jupyterhub-singleuser
-  - jupyterlab=3
+  - jupyterlab
   - jupytext
   - pip
-  - python=3.10
```
Comment (scottyhq) [inline conda/environment.yml:17]: unpinned major dependencies here (since it's been a while this takes us to jlab=4, jbook=1, python=3.12)!

**Comment by scottyhq** [comment]
I also switched the branch protection rule to only require a passing build on Linux (not both linux and osx): 

Screenshot 2024-03-15 at 8 28 13 AM

File: conda/environment.yml (hunk 1 above)
Comment (jomey) [reply conda/environment.yml:17]: I think we froze those versions back then cause the build was breaking with the 0.14 book version.
Generally I am a fan of pinning major versions (i.e <2.0 for Jupyter Book for instance). Brings less surprises when coming back to this template and having things break with libraries advancing their development.

**Comment by jomey** [comment]
Should we try to address #152 with this one too?

**Comment by scottyhq** [comment]
> Should we try to address https://github.com/uwhackweek/jupyterbook-template/issues/152 with this one too? 

The preview looks good so I say we stick with this and revisit 152 seeparately

File: conda/environment.yml (hunk 1 above)
Comment (scottyhq) [reply conda/environment.yml:17]: If there is a problem building in the future we can always bring back a pin. I think no pins are better for solving the full environment dependency tree and ensuring we have the latest functionality, but I don't feel particularly strongly... so happy to pin major versions.

Comment (jomey) [reply conda/environment.yml:17]: How about we just pin the book to be <2 and call it good?

**Pull request merged**